sklearn/manifold/t_sne.py

```diff
@@ -746,8 +761,15 @@ def _fit(self, X, skip_num_points=0):
                 print("[t-SNE] Computing {} nearest neighbors...".format(k))
 
             # Find the nearest neighbors for every point
-            knn = NearestNeighbors(algorithm='auto', n_neighbors=k,
-                                   metric=self.metric)
+            if self.metric_params is None or 'p' not in self.metric_params:
```

Comment:
Can't you use the local variable metric_params_ here to avoid the None condition? Can't you just use `p = metri_params_.pop('p', 2)`?

Reply:
It was my original thought too. But it would consequentially mean hard-coding default value of `p` _outside_ the `NearestNeighbor` constructor. So I suspected it's a bad practice. But if you think it's fine, I'll change it back.